05.ls/ls.rb

```diff
@@ -30,19 +30,19 @@ def output_total_block_count(files)
   puts "total #{total_block_count}"
 end
 
-def output_permission(stat_file)
-  permission_count = stat_file.mode.to_s(8).slice(-3, 3)
-  permission_type = permission_count.split('').map do |file_count|
+def rewrite_permission(stat_file)
+  permission_value = stat_file.mode.to_s(8).slice(-3, 3)
+  permission_types = permission_value.each_char.map do |file_count|
     PERMISSION_TYPE[file_count]
   end
-  permission_type.join('').to_s.ljust(8)
+  permission_types.join('')
```

Comment:
前回もコメントしましたが、

```suggestion
  permission_types.join
```

で同じ結果が得られるはずですー

Reply:
ちなみにめっちゃ高等テクニックですが、

```ruby
permission_value.gsub(/\d/, PERMISSION_TYPE)
```

だけで置き換えることもできたりします。（これは修正しなくてもいいです）

https://docs.ruby-lang.org/ja/latest/method/String/i/gsub.html

Reply:
わー！
`permission_types = permission_value.each_char.map do |file_count|
    PERMISSION_TYPE[file_count]
  end
  permission_types.join('')`が
`permission_value.gsub(/\d/, PERMISSION_TYPE)`一行で置き換えられました！！
教えて下さりありがとうございますー！！